pkg/util/k8s.go

```diff
@@ -30,6 +30,47 @@ func PodStartedBefore(p, q *corev1.Pod) bool {
 	return p.Status.StartTime.Before(q.Status.StartTime)
 }
 
+// TryPodPreferredAZ returns the desired availability zone of the Pod, if it has one
+func TryPodPreferredAZ(pod *corev1.Pod) string {
```

Comment:
The naming here that starts with "try" is a little bit confusing. If you want to emphasize that the preferred AZ might not exist, might it would be better to rename to `PodPreferredAZIfPresent`.

Reply:
Fair enough, makes sense - done